src/main/java/it/uniroma3/mat/extendedset/intset/ImmutableConciseSet.java

```diff
@@ -352,9 +399,12 @@ public int compare(WordHolder h1, WordHolder h2)
 
           i.advanceTo(itr.wordsWalked);
           if (i.hasNext()) {
-            wordsToAdd.add(new WordHolder(i.next(), i));
+            i.word = i.next();
+            qIndex++;
+          } else {
+            removeElement(theQ, qIndex, qSize);
```

Comment:
Don't want to introduce a class (= indirection) here. Actually this PR *removes* classes and makes code more verbose and harder to support. And more bug-prone. But faster.

> For additional optimization it could handle the start index of the queue also (so that you wouldn't have to call System.arraycopy).

Didn't understand this